keps/sig-architecture/617-improve-kep-implementation/README.md

```diff
@@ -0,0 +1,240 @@
+# KEP-617: Enhance the KEP implementation
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [1) Standardize an intra-KEP denotation for &quot;not resolved&quot;](#1-standardize-an-intra-kep-denotation-for-not-resolved)
+  - [2) Change the names of KEPs](#2-change-the-names-of-keps)
+  - [3) Move KEPs from flat-files to a directory structure](#3-move-keps-from-flat-files-to-a-directory-structure)
+  - [4) Metadata split from KEP text](#4-metadata-split-from-kep-text)
+  - [5) Remove some fields from metadata](#5-remove-some-fields-from-metadata)
+  - [6) Formalize one-enhancement == one-KEP](#6-formalize-one-enhancement--one-kep)
+  - [7) Adjust GitHub tooling for KEPs](#7-adjust-github-tooling-for-keps)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
```

Comment:
Could the KEP template be enhanced to have a placeholder for the actual link to the enhancement issue and the current (or future version) of `pull-kubernetes-verify` could validate that there is an issue? For the 1.18 release,  this [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20200123-client-go-ctx.md) got merged pretty quickly and the enhancements team missed it. There was no enhancement issue opened, so we had not been tracking it until a code freeze exception was filed. It easily would have passed enhancement freeze (it checked the criteria), but there was no issue filed. If we that metadata in here so we could do more enforcement before merging, it would have helped catch that I think?

Reply:
If we require a KEP number, isn't that enough to link back to enhancement?

Reply:
Yeah, I saw that requirement when I got further down 👍